### PR TITLE
refactor(optimizer): use `match` to replace `if else`

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

### CODE REVIEW

File: src/query/service/src/sql/planner/optimizer/rule/rewrite/rule_eliminate_filter.rs [hunk 1]
```diff
@@ -56,8 +60,34 @@ impl Rule for RuleEliminateFilter {
...
+        let filter: Filter = s_expr.plan().clone().try_into()?;
+        let mut predicates = Vec::with_capacity(filter.predicates.len());
+        // Delete the predicate if the predicate is `a = a` syntax.
+        for predicate in filter.predicates.iter() {
+            if let Scalar::ComparisonExpr(ComparisonExpr {
+                left, right, op, ..
+            }) = predicate
+            {
+                if op == &ComparisonOp::Equal {
+                    if let Scalar::BoundColumnRef(BoundColumnRef {
```
Comment (sundy-li) [inline src/query/service/src/sql/planner/optimizer/rule/rewrite/rule_eliminate_filter.rs:72]: can we use match pattern here ?

match (left, right) {
  ...
}

**Comment by xudong963** [comment]
Emm, after thinking, I realized `a=a` isn't always true, such as `null=null`.

However, @sundy-li 's suggestion is very good, so I will only change the code style in the ticket.

File: src/query/service/src/sql/planner/optimizer/heuristic/decorrelate.rs [hunk 2]
```diff
@@ -821,28 +821,22 @@ impl SubqueryRewriter {
                 }) = predicate
                 {
                     if op == &ComparisonOp::Equal {
-                        if let Scalar::BoundColumnRef(BoundColumnRef {
-                            column: left_column,
-                        }) = &**left
+                        if let (Scalar::BoundColumnRef(left), Scalar::BoundColumnRef(right)) =
+                            (&**left, &**right)
```
Comment (Xuanwo) [inline src/query/service/src/sql/planner/optimizer/heuristic/decorrelate.rs:825]: Rust sucks.

**Review by Xuanwo — APPROVED** [review]

**Comment by xudong963** [comment]
https://github.com/datafuselabs/databend/actions/runs/3278340663/jobs/5397058710 cc @drmingdrmer

**Comment by drmingdrmer** [comment]
> https://github.com/datafuselabs/databend/actions/runs/3278340663/jobs/5397058710 cc @drmingdrmer

Right I fount this issue too and have been fixed in 3e33b997ad8f6e3304e71edc273ff58c5bc42774

Thanks man!

**Pull request merged**